slackalerts/main.tf

```diff
@@ -0,0 +1,77 @@
+/**
+ * Lambda Function:
+ */
+resource "aws_lambda_function" "sns_to_slack" {
+  filename         = "${path.module}/dist/slackalerts.zip"
+  source_code_hash = filebase64sha256("${path.module}/dist/slackalerts.zip")
+  function_name    = var.name
+  handler          = "index.handler"
+  role             = aws_iam_role.lambda.arn
+  runtime          = "nodejs8.10"
+  environment {
+    variables = {
+      SLACK_TOKEN     = var.SLACK_TOKEN
+      DEFAULT_CHANNEL = var.default_channel
+      TOPIC_MAP       = jsonencode(var.topic_map)
+    }
+  }
+  tags = merge(
+    var.tags,
+    {
+      "Name" = var.name
+    },
+  )
+}
+
+resource "aws_sns_topic_subscription" "default" {
+  count     = var.sns_topic_count
+  endpoint  = aws_lambda_function.sns_to_slack.arn
+  protocol  = "lambda"
+  topic_arn = element(var.sns_topics, count.index)
```

Comment:
I think you did the right thing here in keeping this change as minimal as possible.  Eventually though, I'd love to get rid of the `sns_topics` and `sns_topic_count`, and `default_channel` variables entirely, and just rely on `topic_map`.  Eg:
```hcl
locals {
  sns_topic_arns = [for topic in var.topic_map: topic.arn]
}

resource "aws_sns_topic_subscription" "default" {
  count     = length(local.sns_listen_topics)
  endpoint  = aws_lambda_function.sns_to_slack.arn
  protocol  = "lambda"
  topic_arn = element(local.sns_topic_arns, count.index)
}
```
That way, you'd use that `topic_map` variable to control both the SNS topics we're listening to, and the Slack settings for the topic.  Simple, explicit, and clean. But again - not a change we need to make right now.

Reply:
Ah! I wasn't sure what to do with these, but I'll leave a comment in the ticket and we can make a follow-up if that makes sense?